src/inference_service/session_manager.py

```
@@ -11,36 +13,46 @@
         self.sessions.pop(session.session_id, None)
 
     def get_session_by_id(self, session_id: str):
+        """Look up and return a session by ID, or None if not found."""
         return self.sessions.get(session_id)
 
     def remove_session_by_id(self, session_id: str):
+        """Remove a session from the registry by its session ID."""
         self.sessions.pop(session_id, None)
 
     def get_domain_expert_session(
         self, session_id: str = None
```

Ruff (0.15.7)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)